src/Tests/Microsoft.Diagnostics.Monitoring.Tool.UnitTests/HostBuilderExtensions.cs

```diff
@@ -0,0 +1,111 @@
+
+using Microsoft.Diagnostics.Tools.Monitor;
+using Microsoft.Extensions.Configuration;
+using Microsoft.Extensions.Configuration.EnvironmentVariables;
+using Microsoft.Extensions.Configuration.Memory;
+using Microsoft.Extensions.Hosting;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Reflection;
+using Xunit;
+
+namespace Microsoft.Diagnostics.Monitoring.Tool.UnitTests
+{
+    internal static class HostBuilderExtensions
+    {
+        /// <summary>
+        /// Replaces the ASPNETCORE_* environment configuration source with an in-memory configuration source.
+        /// </summary>
+        public static IHostBuilder ReplaceAspnetEnvironment(this IHostBuilder builder, IDictionary<string, string> values = null)
+        {
+            return builder.ConfigureHostConfiguration(builder =>
+            {
+                ReplaceSource(builder.Sources, IsAspNetChainedConfigurationSource, values);
+            });
+
+            static bool IsAspNetChainedConfigurationSource(IConfigurationSource source)
+            {
+                // ASP.NET injects a precreated configuration source via AddConfiguration during host
+                // configuration. This shows up as a chained configuration source when enumerating the
+                // sources. Use a heuristic to find the chained configuration source added by ASP.NET.
+                return source is ChainedConfigurationSource chainedSource &&
+                    chainedSource.Configuration is ConfigurationRoot chainedConfiguration &&
+                    chainedConfiguration.Providers.Any(p => IsEnvironmentConfigurationProvider(p, "ASPNETCORE_"));
+            }
+
+            static bool IsEnvironmentConfigurationProvider(IConfigurationProvider provider, string prefix)
+            {
+                if (!(provider is EnvironmentVariablesConfigurationProvider envProvider))
+                {
+                    return false;
+                }
+
+                FieldInfo prefixField = typeof(EnvironmentVariablesConfigurationProvider)
+                    .GetField("_prefix", BindingFlags.Instance | BindingFlags.NonPublic);
+                Assert.NotNull(prefixField);
+
+                return string.Equals(prefixField.GetValue(envProvider) as string, prefix, StringComparison.OrdinalIgnoreCase);
+            }
+        }
+
+        /// <summary>
+        /// Replaces the DOTNET_* environment configuration source with an in-memory configuration source.
+        /// </summary>
+        public static IHostBuilder ReplaceDotnetEnvironment(this IHostBuilder builder, IDictionary<string, string> values = null)
+        {
+            return builder.ConfigureHostConfiguration(builder =>
+            {
+                ReplaceEnvironment(builder.Sources, prefix: "DOTNET_", values);
+            });
+        }
+
+        /// <summary>
+        /// Replaces the DOTNETMONITOR_* environment configuration source with an in-memory configuration source.
```

Comment:
I'm pretty sure the capitalization doesn't matter, but pointing out for consistency: I noticed that we say DOTNETMONITOR here, as well as DOTNET in the prior method, but `HostBuilderHelper.ConfigPrefix` is actually set to DotnetMonitor.

Reply:
Correct, capitalization doesn't matter for the prefix. See https://github.com/dotnet/runtime/blob/f85ea976f81945ea18cd5dc71959cccecdc93cd2/src/libraries/Microsoft.Extensions.Configuration.EnvironmentVariables/src/EnvironmentVariablesConfigurationProvider.cs#L48